### PR TITLE
New typings for ServiceNow javascript api reference

### PR DESCRIPTION
**Pull request by bryceg**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`

### CODE REVIEW

File: types/servicenow/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,587 @@
...
+
+declare namespace servicenow {
+    interface GlideRecord {
+        readonly sys_created_by: string & GlideElement;
+        readonly sys_created_on: GlideDateTime & GlideElement;
+        readonly sys_id: string;
+        readonly sys_mod_count: number & GlideElement;
+        readonly sys_updated_by: string & GlideElement;
+        readonly sys_updated_on: GlideDateTime & GlideElement;
+        new (type: string): GlideRecord;
```
Comment (DanielRosenwasser) [inline types/servicenow/index.d.ts]: So `GlideRecord` describes the static side of a class, which when constructed returns... a `GlideRecord`? Are you sure?

File: types/servicenow/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,587 @@
...
+    writeBase64(record: servicenow.GlideRecord, fileName: string, contentType: string, base64Content: string): string;
+    writeContentStream(record: servicenow.GlideRecord, fileName: string, contentType: string, inputStream: object): string;
+}
+
+declare namespace servicenow {
+    interface GlideRecord {
+        readonly sys_created_by: string & GlideElement;
+        readonly sys_created_on: GlideDateTime & GlideElement;
+        readonly sys_id: string;
+        readonly sys_mod_count: number & GlideElement;
```
Comment (DanielRosenwasser) [inline types/servicenow/index.d.ts]: How is this type possible?

File: types/servicenow/tslint.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{ 
+    "extends": "dtslint/dt.json",
+    "rules": {
+        // Interface is needed for merging which is not possible on delcared classes
```
Comment (DanielRosenwasser) [inline types/servicenow/tslint.json]: declared

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/servicenow/index.d.ts (hunk 2 above)
Comment (DanielRosenwasser) [reply types/servicenow/index.d.ts]: Is this the same situation as above?

**Comment by DanielRosenwasser** [comment]
Just as a heads up, your commits don't seem to be associated with your GitHub account. While this isn't technically a problem, you might care if you want more appropriate attribution. You can either make sure your GitHub account is associated with the email address you're using for your commits, or rebase and amend your commits to fix the author name and email.

File: types/servicenow/index.d.ts (hunk 2 above)
Comment (bryceg) [reply types/servicenow/index.d.ts]: Yes, its really a GlideElement but you can access it as its underlying type also

File: types/servicenow/tslint.json (hunk 3 above)
Comment (bryceg) [reply types/servicenow/tslint.json]: fixed

**Pull request merged**